### PR TITLE
Extract pasta_curves crate from halo2 crate

### PR DESCRIPTION
**Pull request by str4d**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -1,19 +1,17 @@
...
     "Ying Tong Lai <yingtong@electriccoin.co>",
     "Daira Hopwood <daira@electriccoin.co>",
     "Jack Grigg <jack@electriccoin.co>",
 ]
 edition = "2018"
-description = """
-Fast proof-carrying data implementation with no trusted setup
-"""
 license = "TGPPL"
-repository = "https://github.com/Electric-Coin-Company/halo2"
```
Comment (daira) [inline Cargo.toml:15]: We should fix this in the `halo2` repo.

File: book/src/design/implementation/fields.md [hunk 2]
```diff
@@ -1,16 +1,16 @@
 # Fields
 
 The [Pasta curves](https://electriccoin.co/blog/the-pasta-curves-for-halo-2-and-beyond/)
-that we use in `halo2` are designed to be highly 2-adic, meaning that a large $2^S$
+are designed to be highly 2-adic, meaning that a large $2^S$
 [multiplicative subgroup](../../background/fields.md#multiplicative-subgroups) exists in
```
Comment (daira) [inline book/src/design/implementation/fields.md:5]:
```suggestion
[multiplicative subgroup](https://zcash.github.io/halo2/background/fields.html#multiplicative-subgroups) exists in
```

File: book/src/design/implementation/fields.md [hunk 3]
```diff
@@ -1,16 +1,16 @@
...
 each field. That is, we can write $p - 1 \equiv 2^S \cdot T$ with $T$ odd. For both Pallas
 and Vesta, $S = 32$; this helps to simplify the field implementations.
 
 ## Sarkar square-root algorithm (table-based variant)
 
 We use a technique from [Sarkar2020](https://eprint.iacr.org/2020/1407.pdf) to compute
-[square roots](../../background/fields.md#square-roots) in `halo2`. The intuition behind
-the algorithm is that we can split the task into computing square roots in each
+[square roots](../../background/fields.md#square-roots) in `pasta_curves`. The intuition
```
Comment (daira) [inline book/src/design/implementation/fields.md:12]:
```suggestion
[square roots](https://zcash.github.io/halo2/background/fields.html#square-roots) in `pasta_curves`. The intuition
```

**Review by daira — APPROVED** [review]
utACK, with a couple of suggested link fixes.

File: book/src/design/implementation/fields.md (hunk 3 above)
Comment (daira) [reply book/src/design/implementation/fields.md:12]: There's another broken link on line 18.

**Review by ebfull — APPROVED** [review]

**Pull request merged**